src/Apps/Auction/__tests__/routes.test.ts

```diff
@@ -24,84 +28,208 @@ describe("Auction/routes", () => {
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/confirm-registration`
+      )
+    })
 
-  it("redirects to the auction registration modal if the user has a qualified credit card", async () => {
-    const { redirect } = await render(
-      `/auction-registration/${Fixture.sale.id}`,
-      mockResolver({
-        ...Fixture,
-        me: {
-          ...Fixture.me,
-          has_qualified_credit_cards: true,
+    it("redirects to sale artwork page if the sale is closed", async () => {
+      const fixture: BidQueryResponse = mockConfirmBidResolver({
+        sale: {
+          is_closed: true,
         },
       })
-    )
-
-    expect(redirect.url).toBe(`/auction/${Fixture.sale.id}/registration-flow`)
-  })
+      const { redirect } = await render(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/bid/${
+          fixture.artwork.id
+        }`,
+        fixture
+      )
+      expect(redirect.url).toBe(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/artwork/${
+          fixture.artwork.id
+        }`
+      )
+    })
 
-  it("redirects back to the auction if the registration window has closed", async () => {
-    const { redirect } = await render(
-      `/auction-registration/${Fixture.sale.id}`,
-      mockResolver({
-        ...Fixture,
+    it("redirects to the sale artwork page if user is not registered and registration is closed", async () => {
+      const fixture: BidQueryResponse = mockConfirmBidResolver({
         sale: {
-          ...Fixture.sale,
+          registrationStatus: null,
           is_registration_closed: true,
         },
       })
-    )
-
-    expect(redirect.url).toBe(`/auction/${Fixture.sale.id}`)
-  })
+      const { redirect } = await render(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/bid/${
+          fixture.artwork.id
+        }`,
+        fixture
+      )
+      expect(redirect.url).toBe(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/artwork/${
+          fixture.artwork.id
+        }`
+      )
+    })
 
-  it("redirects to the auction confirm registration route if bidder has already registered", async () => {
-    const { redirect } = await render(
-      `/auction-registration/${Fixture.sale.id}`,
-      mockResolver({
-        ...Fixture,
+    it("redirects to the registration page if user is not registered and registration is open", async () => {
+      const fixture: BidQueryResponse = mockConfirmBidResolver({
         sale: {
-          ...Fixture.sale,
-          registrationStatus: {
-            qualified_for_bidding: true,
-          },
+          registrationStatus: null,
+          is_registration_closed: false,
         },
       })
-    )
+      const { redirect } = await render(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/bid/${
+          fixture.artwork.id
+        }`,
+        fixture
+      )
+      expect(redirect.url).toBe(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/registration-flow`
+      )
```

Comment:
This was a case that I wasn't sure how to handle. Currently the user would see the registration form right below their bid select component on the confirm bid page but we haven't rebuilt this part yet and it's currently ticketed as separate work. Maybe it makes sense to do both cases at once.

Another option would be to implement some post-registration `?redirectTo` logic in our `Register` reaction app so that we can send them to the bid form after they have successfully registered (this sounds like it would get very complicated fast to me).

Reply:
I think that still might be okay. Happy to pair on it. I don't think it's blocking this PR.